internal/pkg/deploy/pipeline.go

```diff
@@ -67,93 +70,100 @@ func (a *ArtifactBucket) Region() (string, error) {
 	return parsedArn.Region, nil
 }
 
-// Source defines the source of the artifacts to be built and deployed.
-type Source struct {
-	// The name of the source code provider. For example, "GitHub"
-	ProviderName string
+// GitHubSource defines the (GH) source of the artifacts to be built and deployed.
+type GitHubSource struct {
+	ProviderName                string
```

Comment:
Is this the same struct which determines the values that get injected into CFN? Is that why we need the `ProviderName` struct member?

Reply:
Yeah--I had taken it out because it seemed redundant, but this seemed to be the most straightforward way to determine which chunk of the yaml should get applied to the template. 

I may change this when I introduce other sources and partials. Let me know if you have ideas!

Reply:
We could create a base `pipelineSource` struct which has a `ProviderName()` method and include that as a member of both `GithubSource` and `CodeCommitSource`, as something like 

```go
type pipelineSource struct {
    providerName string
}

func (s *pipelineSource) ProviderName() string {
    return s.providerName
}

type GithubSource struct {
    *pipelineSource
    // other GH specific members
}

func NewGithubSource(url, branch, secretID string) GithubSource {
    return GithubSource{
        pipelineSource: &pipelineSource{
            providerName: manifest.GithubProviderName
        },
        // other members
    }
}
```

Reply:
This is cool! I implemented it locally but didn't commit because it may overcomplicate things at this point (unless there's a benefit to the cfn template calling a method rather than a struct field). I'll keep this in mind, though, if we have more functionality to add to this set-up with more sources.

Reply:
There's no particular benefit except 
a) avoiding the confusion between members and methods in the template package
b) making it harder to incorrectly initialize `GithubSource` or `CodeCommitSource` with the wrong provider name. 

If the struct becomes more complicated then I think it would be worth it to move to the constructor pattern above but for the time being it's totally fine to leave as is.